### PR TITLE
fix(medusa-payment-stripe): handle webhook sirialization failure

### PR DESCRIPTION
**Pull request by adrien2p**
WIP: ready only for snap

**What**

Handle stripe webhook serialization failure on `payment_intent.amount_capturable_updated` event

FIXES CORE-796

### CODE REVIEW

File: packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js [hunk 1]
```diff
@@ -34,12 +37,32 @@ export default async (req, res) => {
...
+              err?.detail ?? err?.message
+            }`
+            if (err.code === PostgresError.SERIALIZATION_FAILURE) {
+              message = `Stripe webhook ${
+                event.type
+              } handle failed. This can happen when this webhook is triggered during a cart completion and can be ignored. This event should be retried automatically.\n${
+                err?.detail ?? err?.message
+              }`
+            }
+            logger.warning(message)
```
Comment (srindom) [inline packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js:62]: thought(non-blocking): this seems like a lot of work to put on the shoulders of the plugin author; would love for us to figure out a better way to handle payments so it becomes easier to build plugins that are reliable and robust. 

--
There is no action expected from my side - just an observation :)

Comment (adrien2p) [reply packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js:62]: It seams that with the stripe webhook and serializable transaction we don’t have much choice for now. When the payment will be re worked and maybe they will be async, we will probably have the same kind of behaviour to manage as we will probably want the payment provider work to be serialisable and retriable. Here the intent is to let stripe try again later. Also, i think that as a plugin author working with payment provider we can’t avoid that much to expect them to manage race condition such as a webhook triggered while doing some other actions, do you have something in mind?

**Comment by adrien2p** [comment]
@srindom could you give me your opinion on the webhook handler method I have created to wrap the logic under the hood so that a plugin creator does not have to handle it. I am not sure about putting it in the payment provider but since it is our link to the payment provider plugin I thought it would make sense. Let me know.

**Comment by adrien2p** [comment]
Testsd by a user as well https://discord.com/channels/876835651130097704/1041787790297419786/1042211674414915654

**Comment by adrien2p** [comment]
@olivermrbl would you happen to know why the database pipeline is failing?

File: packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js (hunk 1 above)
Comment (srindom) [reply packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js]: I would rather see that we rever to this version and rethink this flow separately than the proposal you have in the latest commits;

Comment (adrien2p) [reply packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js]: Alright, i ll revert it

Comment (adrien2p) [reply packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js]: done

File: packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import { PostgresError } from "@medusajs/medusa/src/utils"
```
Comment (olivermrbl) [inline packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js:1]: comment: Should this be exported from index at root so we can import it like so: `import {} from @medusajs/medusa`?

Comment (olivermrbl) [reply packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js:1]: non-blocking

Comment (adrien2p) [reply packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js:1]: yes defiantly , I was not sure about it but if you agree then yes

File: packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js [hunk 3]
```diff
@@ -10,13 +12,14 @@ export default async (req, res) => {
     return
   }
 
+  const paymentProviderService = req.scope.resolve("paymentProviderService")
```
Comment (olivermrbl) [inline packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js:15]: todo: Not used so we can remove

File: packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js [hunk 4]
```diff
@@ -90,3 +103,23 @@ export default async (req, res) => {
     await handleCartPayments(event, req, res, cartId ?? resourceId)
   }
 }
+
+function paymentIntentAmountCapturableEventHandler({
```
Comment (olivermrbl) [inline packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js:107]: todo: Shoud be `async`

File: packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js [hunk 5]
```diff
@@ -90,3 +103,23 @@ export default async (req, res) => {
     await handleCartPayments(event, req, res, cartId ?? resourceId)
   }
 }
+
```
Comment (olivermrbl) [inline packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js:106]: question: My IDE complains that `paymentIntentId` is undefined on line 101. We should use `paymentIntent.id`. Do you get the same?

**Review by olivermrbl — CHANGES_REQUESTED** [review]

File: packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js (hunk 4 above)
Comment (adrien2p) [reply packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js:105]: no it should not 👍

File: packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js (hunk 5 above)
Comment (adrien2p) [reply packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js:107]: I don't get anything, let me see

Comment (adrien2p) [reply packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js:107]: ok it was not shown as an error cause it is js i ve fixed it

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**